### PR TITLE
`Iris`: Add tiered settings system

### PR DESCRIPTION
**Pull request by Hialus**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/guidelines/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [x] I added pre-authorization annotations according to the [guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/#rest-endpoint-best-practices-for-authorization) and checked the course groups for all new REST Calls (security).
- [x] I documented the Java code using JavaDoc style.
#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Description

I added a settings system for Iris consisting of three tiers: Global, Course, and Programming Exercise.
- The global settings apply for all of Artemis and can be edited by Admins
- The course settings apply for an entire course and can be edited by Instructors
- The programming exercise settings apply for a single programming exercise and can be edited by Editors

Each tier has identical settings consisting of two sub settings for the chat and hestia sub features.
Each sub feature has 3 different settings that you can edit:
- Enabled - If this feature is enabled. Must be true on all 3 tiers to be true
- Preferred Model - The preferred LLM to use. Not fully implemented yet. Will be done in a future PR
- Template - The guidance template to use. Can be inherited from higher tiers if needed

There is one exception: The Hestia subfeature does not have to be specifically enabled for the programming exercise if it is already enabled on a global and course wide level. In this case the settings are inherited and the feature is available without further configuration.

**Some more notes:**
- By default all features are disabled on a global level.
- The default templates will be added in a future PR.
- The UI is mostly a prototype so far and will be improved in the future.

### Steps for Testing

Prerequisites:
- 1 Admin
- 1 Course
- 1 Java Programming Exercise with coverage enabled
- TS9

This is one possible path you can take. There are a lot more and this also contains some alternate paths already.
1. Log in to Artemis
2. Open the admin drop down
3. See the new Iris entry at the bottom and click it
4. Here you can now enable/disable the two subfeature globally. If you change this, make sure no one else is testing this right now, as this will interfere with their testing
5. If you want change the templates
6. Save
7. Edit the templates again
8. Reload
9. Templates should be reset
10. Go to the course management page
11. See the new Iris button and click it
12. Enable the sub features
13. Save
14. Change one of the templates to inherited/not inherited
15. Save
16. Go to the programming exercise page
17. See the iris button and click it
18. Set the Hestia feature to inherited/not inherited
19. If not inherited: Enable/Disable it
20. Save
21. Go to the hints menu
22. Go to the generate hints menu
23. Go through the wizard to generate code hints
24. Edit one of the code hints
25. If you have hestia enabled on a global and course level, as well as having it inherited or enabled on the PE level there should be a button to generate descriptions. If it is disable on any tier, the button will not be there. Please enable it to continue
26. Click the button if available. Descriptions should be generated
27. Disable it in any tier you want
28. Go back to the edit code hint menu
29. The button is gone
30. Make sure that the chat feature is enable on all three tiers
31. Participate in the exercise
32. You should see the chat icon in the bottom right corner
33. Disable the chate feature in any tier
34. Go back to the exercise participation page
35. The chat button is gone

#### Performance Review
- [x] I confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [x] I confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

#### Client

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [course-management-tab-bar.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4990-12/artifact/shared/Coverage-Report-Client-Tests/app/course/manage/course-management-tab-bar/course-management-tab-bar.component.ts.html) | 92.75% | ✅ |
| [iris-settings.model.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4990-12/artifact/shared/Coverage-Report-Client-Tests/app/entities/iris/settings/iris-settings.model.ts.html) | 100% | ✅ |
| [iris-sub-settings.model.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4990-12/artifact/shared/Coverage-Report-Client-Tests/app/entities/iris/settings/iris-sub-settings.model.ts.html) | 100% | ✅ |
| [iris-template.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4990-12/artifact/shared/Coverage-Report-Client-Tests/app/entities/iris/settings/iris-template.ts.html) | 100% | ✅ |
| [exercise-hint-update.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4990-12/artifact/shared/Coverage-Report-Client-Tests/app/exercises/shared/exercise-hint/manage/exercise-hint-update.component.ts.html) | 82.5% | ✅ |
| [iris-course-settings-update.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4990-12/artifact/shared/Coverage-Report-Client-Tests/app/iris/settings/iris-course-settings-update/iris-course-settings-update.component.ts.html) | 75% | ✅ |
| [iris-programming-exercise-settings-update.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4990-12/artifact/shared/Coverage-Report-Client-Tests/app/iris/settings/iris-programming-exercise-settings-update/iris-programming-exercise-settings-update.component.ts.html) | 75% | ✅ |
| [iris-settings-update.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4990-12/artifact/shared/Coverage-Report-Client-Tests/app/iris/settings/iris-settings-update/iris-settings-update.component.ts.html) | 82% | ✅ |
| [iris-sub-settings-update.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4990-12/artifact/shared/Coverage-Report-Client-Tests/app/iris/settings/iris-settings-update/iris-sub-settings-update/iris-sub-settings-update.component.ts.html) | 100% | ✅ |
| [iris-settings.service.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4990-12/artifact/shared/Coverage-Report-Client-Tests/app/iris/settings/shared/iris-settings.service.ts.html) | 100% | ✅ |
| [course-exercise-details.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4990-12/artifact/shared/Coverage-Report-Client-Tests/app/overview/exercise-details/course-exercise-details.component.ts.html) | 83.72% | ✅ |
| [navbar.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4990-12/artifact/shared/Coverage-Report-Client-Tests/app/shared/layouts/navbar/navbar.component.ts.html) | 90.94% | ✅ |

#### Server

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [Course.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4990-7/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.domain/Course.html) | 89% | ✅ |
| [ProgrammingExercise.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4990-7/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.domain/ProgrammingExercise.html) | 92% | ✅ |
| [IrisChatSession.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4990-7/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.domain.iris.session/IrisChatSession.html) | 83% | ✅ |
| [IrisHestiaSession.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4990-7/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.domain.iris.session/IrisHestiaSession.html) | 86% | ✅ |
| [IrisMessage.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4990-7/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.domain.iris/IrisMessage.html) | 94% | ✅ |
| [IrisSession.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4990-7/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.domain.iris.session/IrisSession.html) | 100% | ✅ |
| [IrisTemplate.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4990-7/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.domain.iris/IrisTemplate.html) | 95% | ✅ |
| [IrisSettings.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4990-7/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.domain.iris.settings/IrisSettings.html) | 100% | ✅ |
| [IrisSubSettings.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4990-7/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.domain.iris.settings/IrisSubSettings.html) | 100% | ✅ |
| [IrisModel.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4990-7/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.connectors.iris/IrisModel.html) | 100% | ✅ |
| [IrisMessageService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4990-7/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.iris/IrisMessageService.html) | 100% | ✅ |
| [IrisModelService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4990-7/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.iris/IrisModelService.html) | 73% | ✅ |
| [IrisSessionService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4990-7/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.iris/IrisSessionService.html) | 90% | ✅ |
| [IrisSettingsService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4990-7/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.iris/IrisSettingsService.html) | 88% | ✅ |
| [IrisWebsocketService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4990-7/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.iris/IrisWebsocketService.html) | 88% | ✅ |
| [IrisGPT3_5Service.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4990-7/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.iris.model/IrisGPT3_5Service.html) | 90% | ✅ |
| [IrisChatSessionService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4990-7/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.iris.session/IrisChatSessionService.html) | 96% | ✅ |
| [IrisHestiaSessionService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4990-7/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.iris.session/IrisHestiaSessionService.html) | 89% | ✅ |
| [CodeHintResource.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4990-7/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.web.rest.hestia/CodeHintResource.html) | 86% | ✅ |
| [IrisMessageResource.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4990-7/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.web.rest.iris/IrisMessageResource.html) | 100% | ✅ |
| [IrisSessionResource.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4990-7/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.web.rest.iris/IrisSessionResource.html) | 100% | ✅ |
| [IrisSettingsResource.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS4990-7/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.web.rest.iris/IrisSettingsResource.html) | 92% | ✅ |

### Screenshots

Admin dropdown entry:
grafik
Global settings view:
grafik
Course view button:
grafik
Course settings view:
grafik
Programming exercise button:
grafik
Programming exercise settings:
grafik

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/service/iris/IrisSettingsService.java
Comment (SolizerCodes) [inline src/main/java/de/tum/in/www1/artemis/service/iris/IrisSettingsService.java:62]: Note: orElseThrow() will never throw since settingsOptional.size > 1 and max only returns an empty Optional if the stream was empty

File: src/main/java/de/tum/in/www1/artemis/service/iris/IrisSettingsService.java
Comment (SolizerCodes) [inline src/main/java/de/tum/in/www1/artemis/service/iris/IrisSettingsService.java:231]: Ternary is hard to read here. Maybe consider something like this:
```suggestion
            var preferredModel = null;
            if (subSettings2 != null && subSettings2.getPreferredModel() != null) {
                preferredModel = subSettings2.getPreferredModel();
            } else if (subSettings1 != null) {
                preferredModel = subSettings1.getPreferredModel();
            }
```

File: src/main/java/de/tum/in/www1/artemis/service/iris/IrisSettingsService.java
Comment (SolizerCodes) [inline src/main/java/de/tum/in/www1/artemis/service/iris/IrisSettingsService.java:235]: Or just add braces like this for example:
```suggestion
            var template = (subSettings2 != null && subSettings2.getTemplate() != null) ? subSettings2.getTemplate() : (subSettings1 != null ? subSettings1.getTemplate() : null);
```

**Review by SolizerCodes — COMMENTED** [review]
Looked at the server code and added three little comments

File: src/main/java/de/tum/in/www1/artemis/service/iris/IrisSettingsService.java
Comment (Hialus) [reply src/main/java/de/tum/in/www1/artemis/service/iris/IrisSettingsService.java:231]: Fixed it in #6714.

File: src/main/java/de/tum/in/www1/artemis/service/iris/IrisSettingsService.java
Comment (Hialus) [reply src/main/java/de/tum/in/www1/artemis/service/iris/IrisSettingsService.java:235]: Fixed it in #6714

**Review by SolizerCodes — COMMENTED** [review]
I tested the code and found two things:
1. The toggles in the global Iris settings are very dark when not selected. Is that styleguide? If no they should be a bit lighter imo.
2. Currently, when only looking at a programming exercise, it looks like Iris can be activated for the exercise even if the course has Iris deactivated. I was thinking maybe remove the Iris Settings Button in an exercise if Iris is deactivated course-wide or add a disclaimer or something to indicate that even if you activate Iris for the exercise it will still not be activated due to the course-wide deactivation.

**Comment by Hialus** [comment]
> I tested the code and found two things:
> 
>     1. The toggles in the global Iris settings are very dark when not selected. Is that styleguide? If no they should be a bit lighter IMO.

This is the default look of this component, so I won't add any special styling for now, since this is generally an unfinished UI so far.

>     2. Currently, when only looking at a programming exercise, it looks like Iris can be activated for the exercise even if the course has Iris deactivated. I was thinking maybe remove the Iris Settings Button in an exercise if Iris is deactivated course-wide or add a disclaimer or something to indicate that even if you activate Iris for the exercise it will still not be activated due to the course-wide deactivation.

I guess this is also a good thing to implement in a follow-up, when the UI is improved

**Review by DemjenDavid — DISMISSED** [review]
Tested on ts9, 3 tier enable/disable works correctly

File: src/main/java/de/tum/in/www1/artemis/service/iris/IrisSettingsService.java
Comment (tobias-lippert) [inline src/main/java/de/tum/in/www1/artemis/service/iris/IrisSettingsService.java:32]: I think refactoring is out of scope for this PR, but just you are aware of it

File: src/main/java/de/tum/in/www1/artemis/service/iris/IrisSettingsService.java
Comment (tobias-lippert) [inline src/main/java/de/tum/in/www1/artemis/service/iris/IrisSettingsService.java:273]: could we make this an optional to avoid setting it to null?

File: src/main/java/de/tum/in/www1/artemis/service/iris/IrisSettingsService.java
Comment (tobias-lippert) [inline src/main/java/de/tum/in/www1/artemis/service/iris/IrisSettingsService.java:366]: add braces please

File: src/main/java/de/tum/in/www1/artemis/web/rest/iris/IrisSessionResource.java
Comment (tobias-lippert) [inline src/main/java/de/tum/in/www1/artemis/web/rest/iris/IrisSessionResource.java:67]: doesn't follow our authorization guide lines
should be @EnforceAtLeastStudent

File: src/main/java/de/tum/in/www1/artemis/web/rest/iris/IrisSessionResource.java
Comment (tobias-lippert) [inline src/main/java/de/tum/in/www1/artemis/web/rest/iris/IrisSessionResource.java:86]: see comment above

File: src/main/java/de/tum/in/www1/artemis/web/rest/iris/IrisSettingsResource.java
Comment (tobias-lippert) [inline src/main/java/de/tum/in/www1/artemis/web/rest/iris/IrisSettingsResource.java:25]: unused apparently

File: src/main/java/de/tum/in/www1/artemis/web/rest/iris/IrisSettingsResource.java
Comment (tobias-lippert) [inline src/main/java/de/tum/in/www1/artemis/web/rest/iris/IrisSettingsResource.java:52]: doesn't follow our authorization guidelines
should be @EnforceAtLeastInstructor

File: src/main/java/de/tum/in/www1/artemis/web/rest/iris/IrisSettingsResource.java
Comment (tobias-lippert) [inline src/main/java/de/tum/in/www1/artemis/web/rest/iris/IrisSettingsResource.java:65]: doesn't follow our authorization guidelines.
Should be @EnforceAtLeastEditor

File: src/main/java/de/tum/in/www1/artemis/web/rest/iris/IrisSettingsResource.java
Comment (tobias-lippert) [inline src/main/java/de/tum/in/www1/artemis/web/rest/iris/IrisSettingsResource.java:80]: see comment above

File: src/main/java/de/tum/in/www1/artemis/web/rest/iris/IrisSettingsResource.java
Comment (tobias-lippert) [inline src/main/java/de/tum/in/www1/artemis/web/rest/iris/IrisSettingsResource.java:97]: same as above

File: src/main/java/de/tum/in/www1/artemis/web/rest/iris/IrisSettingsResource.java
Comment (tobias-lippert) [inline src/main/java/de/tum/in/www1/artemis/web/rest/iris/IrisSettingsResource.java:116]: see comment above

File: src/main/java/de/tum/in/www1/artemis/web/rest/iris/IrisSettingsResource.java
Comment (tobias-lippert) [inline src/main/java/de/tum/in/www1/artemis/web/rest/iris/IrisSettingsResource.java:149]: see comments above

File: src/main/java/de/tum/in/www1/artemis/web/rest/iris/IrisSettingsResource.java
Comment (tobias-lippert) [inline src/main/java/de/tum/in/www1/artemis/web/rest/iris/IrisSettingsResource.java:166]: see comments above

File: src/main/webapp/app/admin/admin.route.ts
Comment (tobias-lippert) [inline src/main/webapp/app/admin/admin.route.ts:16]: apparently unused according to github

File: src/main/webapp/app/course/manage/course-management-tab-bar/course-management-tab-bar.component.html
Comment (tobias-lippert) [inline src/main/webapp/app/course/manage/course-management-tab-bar/course-management-tab-bar.component.html:79]: couldn't we use course?. instead of forcing it to be defined?

**Review by tobias-lippert — CHANGES_REQUESTED** [review]
The authorization annotations should be changed.

File: src/main/java/de/tum/in/www1/artemis/domain/ProgrammingExercise.java
Comment (Hialus) [reply src/main/java/de/tum/in/www1/artemis/domain/ProgrammingExercise.java:148]: Not caused by this PR, but by the addition of `public String getType()` by Stephan in another PR.

File: src/main/java/de/tum/in/www1/artemis/domain/iris/settings/IrisSettings.java
Comment (Hialus) [reply src/main/java/de/tum/in/www1/artemis/domain/iris/settings/IrisSettings.java:3]: The editorconfig just sets the amount of imports after which the * import for the package should be used. Teamscale does not use this and probably needs to be adjusted separately.

File: src/main/java/de/tum/in/www1/artemis/service/iris/IrisSettingsService.java
Comment (Hialus) [reply src/main/java/de/tum/in/www1/artemis/service/iris/IrisSettingsService.java:32]: Codacy doesn't like big classes. I guess I can look into splitting it into smaller services in a future PR.

File: src/main/webapp/app/admin/admin.route.ts
Comment (Hialus) [reply src/main/webapp/app/admin/admin.route.ts:16]: Was probably a remnant, when I initially did not lazy load it 🤷🏻

File: src/main/java/de/tum/in/www1/artemis/service/iris/IrisSettingsService.java
Comment (Hialus) [reply src/main/java/de/tum/in/www1/artemis/service/iris/IrisSettingsService.java:273]: IntelliJ then tells me "'One To One' attribute type should not be 'Optional'", so I'd leave it as is for now.

**Comment by Hialus** [comment]
Thanks for your review @tobias-lippert.
Regarding the PreAuthorize annotations: This change is not yet merged into develop-iris, so I can't implement it. Once it is merged from develop into develop-iris, I'll create a separate PR to adjust it for all Iris endpoints.

**Review by SabrinaGlatz — COMMENTED** [review]
tested on ts9, the settings work as expected. I just have a few comments.
-  If you open the Course Iris Settings or Programming Task Iris setting, the overlay with the settings appears but I couldn't find a way to close this again except for clicking next to it. It would be nice if there was another way to close it. If this is a common thing that is often used in Artemis like that and people know how to close it, it's fine.
- Is it okay, that it is possible to disable something on a global level but enable it on a course label (I mean it won't work but it looks in the interface as if it was possible)? I thought the point of having these layers is that if e.g. the global is disabled it won't be able to use it in the courses. I think it would be confusing if editors try to enable it but don't know that it was disabled globally.
- Does Hestia belong to the Iris settings? Maybe some people won't expect it to be there, you could rename the settings "GPT/LLM settings" or something like that and include all current/future applications of GPT/LLM there. 
- I would also add an info icon next to the "Template" to describe what should be added there. If someone has not used this before that might be unclear.
- Also for inheriting the template, an editor who is not an admin might not know what he is inheriting there. Maybe there could be some information regarding the inherited template, so he needs if he needs to modify it. 
- In the German version of the Iris settings "Inherit Template" is still English. Maybe we could change it to "Erbe Template" or something like that but it's not that necessary. However, at the Programming tasks this is German only for Hestia ("Vererbe Hestia Einstellungen") but it should rather be "Erbe" instead of "Vererbe", right?

**Review by SolizerCodes — DISMISSED** [review]
Reapprove

**Review by tobias-lippert — DISMISSED** [review]
Approve after addressing my change requests.

**Comment by Hialus** [comment]
Thanks for your details comment, Sabrina. As stated the UI is more of a prototype right now and not the focus of this PR, so I would address them in a follow-up PR.

**Pull request merged**
